### ISSUE TITLE
Always converging to classyfing everything with one label.

### ISSUE THREAD

**Issue by WJ44** [BrainNetwork/BrainNet#1]
Dear authors,

I've been playing around a bit with the code from your paper and I am struggling to get any promising result. It can sometimes look slightly promising during the start of training, but I can never reach an accuracy above say 30%. When continuing training the network seems to always converge to classifying everything under one label. This also seems to almost always be 0 which seems somewhat strange. I have played around a bit with the options and hyperparameters but nothing I do seems to help much.

This is what I am currently doing:

```
import torch
import torchvision

from DataGenerator import random_halfspace_data
from LocalNetBase import Options, UpdateScheme
from network import LocalNet
from train import train_local_rule, evaluate, train_es

opt = Options(use_input_rule=False,
              use_output_rule=True,
              use_graph_rule=True,
              gd_graph_rule=True,
              gd_output_rule=False,
              gd_input=False,
              gd_output=False,
              additive_rule=True)

upd = UpdateScheme(cross_entropy_loss=True,
                   mse_loss=False,
                   update_misclassified=False,
                   update_all_edges=True)

net = LocalNet(n=784, m=10, num_v=100, p=1, cap=-1, rounds=2, step_sz=0.01, options=opt, update_scheme=upd)


transform = torchvision.transforms.Compose([torchvision.transforms.ToTensor(),
                                            torchvision.transforms.Lambda(lambda x : x.flatten().double())])


trainset = torchvision.datasets.MNIST(root='data', train=True, download=True, transform=transform)

X = trainset.data.flatten(start_dim = 1).numpy()
y = trainset.targets.numpy()

testset = torchvision.datasets.MNIST(root='data', train=False, download=True, transform=transform)

X_test = testset.data.flatten(start_dim = 1).numpy()
y_test = testset.targets.numpy()


net.set_output_rule(torch.tensor([[-1,1],[1,-1],[-1,1],[1,-1],[-1,1],[1,-1],[-1,1],[1,-1]]))

losses, train_accuracies, test_accuracies = train_local_rule(X=X, y=y, meta_model=net, rule_epochs=200, epochs=1, batch=200, lr=0.01, X_test=X_test, y_test=y_test, verbose=True)
```

I hope you can give me some pointers to get me along. I am currently at a bit of a loss.
